docs/js-guide.md

```diff
@@ -0,0 +1,143 @@
+# JavaScript guide
+
+This document explains the tools and conventions that we use to maintain
+quality and consistency of our JavaScript code.
+
+## Language style guide
+
+### Use of language features
+
+Projects should generally prefer native ES language features over equivalent
+idioms from earlier versions of the language (eg. classes over constructor functions,
+`import` over `require(...)`).
+
+_The native features typically enable better tooling._
+
+Projects may use any ES language features which meet the following criteria:
+
+1. Supported by the tooling we use (Babel, ESLint)
+2. Supported natively in the stable release of at least two major browsers.
+
+   _Waiting for the feature to advance far enough to be natively supported
+   reduces maintenance problems due to using language features which are not
+   widely supported or which are revised significantly during the standardization
+   process._
+
+3. Can be transpiled for supported browsers that do not natively support the
+   feature without excessive overhead.
+
+   _Some language features (generators, as a historical example) are difficult
+   to transpile for older browsers without introducing a lot of code that
+   slows startup and makes debugging difficult._
+
+As an exception, [JSX](https://reactjs.org/docs/introducing-jsx.html) is used for
+creating UI components.
+
+### Variable declarations
+
+Variables should be declared with `const` where possible, as this makes it clear
+that the variable is only assigned in one place.
+
+## Formatting code
+
+All of our projects use [Prettier](https://prettier.io) to format JavaScript code,
```

Comment:
I wonder about the maintainability of this chunk of documentation. Will we remember to update it if we change the way we configure prettier? Would a link to an example `prettierrc` in one of our repos do?

Reply:
I think we'll be OK. The only scenario where I envisage deviating additionally from Prettier's defaults is if they make a major change in a new release and we decide that we don't want to roll with it. Otherwise we try to keep the config as minimal as possible.